### PR TITLE
Migration guide for v7 and doc tweaks

### PR DESCRIPTION
**Pull request by damieng**
Provides the migration guide for v7 as well as some minor doc improvements and tweaks for v7.

### CODE REVIEW

File: docs-source/index.md [hunk 1]
```diff
@@ -178,28 +177,27 @@ In your controller:
...
 ### Mocking the ManagementApiClient
 
 While the `AuthenticationApiClient` has an associated `IAuthenticationApiClient` interface you will find that `ManagementApiClient` does not. The reason for this is that the such an interface would encompass the entire management API surface and that any change to the Management API would break existing users of that interface.
 
-Instead, given that the Management API SDK is just a strongly-typed wrapper around the REST API, you can instead mock the @Auth0.ManagementApi.IManagementConnection interface which has just three methods:
-
-1. `void SetDefaultHeaders(IDictionary<string, string> headers)` - You can likely ignore this in your mocks
-2. `Task<T> GetAsync<T>(Uri uri, ...)` - To handle GET requests
-3. `Task<T> SendAsync<T>(HttpMethod method, Uri uri, object body, ...)` - To handle POST/PUT/PATCH/DELETE 
+Instead, given that the Management API SDK is just a strongly-typed wrapper around the REST API, you can instead mock the @Auth0.ManagementApi.IManagementConnection interface which has just two methods:
```
Comment (lbalmaceda) [inline docs-source/index.md]: "Instead" usage

```suggestion
Instead, given that the Management API SDK is just a strongly-typed wrapper around the REST API, you can mock the @Auth0.ManagementApi.IManagementConnection interface which has just two methods:
```

File: docs-source/migrating.md [hunk 2]
```diff
@@ -2,9 +2,7 @@
 
 ## Migrating from v6 to v7
```
Comment (lbalmaceda) [inline docs-source/migrating.md]: Do we want to keep this file around? Are there still people willing to migrate from v4, for example?

File: docs-source/migrating-to-v7.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+JWKS caching has changed. Previously it would cache the document once and it would be indefinitely until the app is restarted.
+
+The JWKS document will now only cache for 10 minutes and will retry more frequently when presented with new signing keys. This provides support for rotation of signing keys in the future.
+
+#### HS256
+
+- If using HS256 and your app is confidential (e.g. a web server) then you must set `SigningAlgorithm` to `SigningAlgorithm.HS256` on all `AuthenticationApiClient` requests.
+
+- If using HS256 and your app is NOT confidential (e.g. a desktop or mobile app) you should plan to move to RS256 as soon as possible. Client secrets can not be kept secure in these types of applications.
```
Comment (lbalmaceda) [inline docs-source/migrating-to-v7.md]: What about adding a link to an article that talks the diffs between confidential / non confidential, like this one https://auth0.com/docs/applications/concepts/app-types-confidential-public

File: docs-source/migrating.md (hunk 2 above)
Comment (damieng) [reply docs-source/migrating.md]: I don't think there is any harm in keeping it around for now.

**Pull request merged**